sql/src/main/java/io/crate/analyze/expressions/SubqueryAnalyzer.java

```diff
@@ -37,7 +37,8 @@ public SubqueryAnalyzer(RelationAnalyzer relationAnalyzer, StatementAnalysisCont
         this.statementAnalysisContext = statementAnalysisContext;
     }
 
-    public AnalyzedRelation analyze(Query query) {
-        return relationAnalyzer.analyze(query, statementAnalysisContext);
+    public QueriedRelation analyze(Query query) {
+        // The only non-queried relations are base tables - which cannot occur as part of a subquery. so this cast is safe.
+        return (QueriedRelation) relationAnalyzer.analyze(query, statementAnalysisContext);
```

Comment:
Not sure if we gain all that much - it just moves the cast to a different place.